common4j/src/main/com/microsoft/identity/common/java/interfaces/IPlatformComponents.java

```diff
@@ -66,11 +64,39 @@
     IDevicePopManager getDefaultDevicePopManager() throws ClientException;
 
     /**
-     * Gets a {@link IDevicePopManager} associated to the key alias.
+     * Retrieve a name-value store with a given identifier.
      *
-     * @param alias the alias of the key. Storage mechanisms can use this alias to provide access.
-     * @throws ClientException if it fails to initalize, or if the operation is not supported by the platform.
+     * @param storeName The name of a new KeyValue store.
+     * @param clazz     The class of values in the name value store.
+     * @return a INameValueStorage instance based around data stored with the same storeName.
      */
+    <T> INameValueStorage<T> getNameValueStore(String storeName, Class<T> clazz);
+
+    /**
+     * Retrieve a name-value store with a given identifier.
+     *
+     * @param storeName The name of a new KeyValue store. May not be null.
+     * @param helper    The key manager for the encryption.  May be null.
+     * @param clazz     The class of values in the name value store. May not be null.
+     * @return a INameValueStorage instance based around data stored with the same storeName.
+     */
+    <T> INameValueStorage<T> getEncryptedNameValueStore(String storeName, IKeyAccessor helper, Class<T> clazz);
+
+    /**
+     * Get a generic encrypted ISharedPreferencesFileManager with a given identifier.
+     *
+     * @param storeName The name of a new KeyValue store. May not be null.
+     * @param helper    The key manager for the encryption.  May not be null.
+     */
+    IKeyBasedStorage getEncryptedFileStore(String storeName, IKeyAccessor helper);
```

Comment:
We might want to create a post-p0 PBI to combine IKeyBasedStorage and INameValueStorage.

(My understanding is that you want to minimize change (as IKeyBasedStorage supports getLong and getString), but internally they should be pretty similar.

Reply:
(and the name IKeyBasedStorage and INameValueStorage... is pretty close. This will confuse people in the long run IMO).

Reply:
+1

Reply:
may be rename `IKeyBasedStorage` to `IMultiTypeNameValueStorage` ?

Reply:
OK.  I think the thing that it hangs on to is the file name.